src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/commands/mobile/publish_app_service_request.cc

```diff
@@ -93,6 +93,41 @@ void PublishAppServiceRequest::Run() {
       application_manager_.GetAppServiceManager().PublishAppService(
           manifest, true, connection_key());
 
+  std::string requested_service_name = "";
+
+  if ((*message_)[strings::msg_params][strings::app_service_manifest].keyExists(
+          strings::service_name)) {
+    requested_service_name =
+        (*message_)[strings::msg_params][strings::app_service_manifest]
+                   [strings::service_name].asString();
+  }
+
+  // Todo: revision make service type string.
+  mobile_apis::AppServiceType::eType requested_service_type =
+      static_cast<mobile_apis::AppServiceType::eType>(
+          (*message_)[strings::msg_params][strings::app_service_manifest]
+                     [strings::service_type].asUInt());
+
+  smart_objects::SmartArray* requested_handled_rpcs;
+  if ((*message_)[strings::msg_params][strings::app_service_manifest].keyExists(
+          strings::handled_rpcs)) {
+    requested_handled_rpcs =
+        (*message_)[strings::msg_params][strings::app_service_manifest]
+                   [strings::handled_rpcs].asArray();
+  }
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  bool result =
+      policy_handler_.CheckAppServiceParameters(app->policy_app_id(),
+                                                requested_service_name,
+                                                requested_service_type,
+                                                requested_handled_rpcs);
+
+  if (!result) {
+    SendResponse(false, mobile_apis::Result::DISALLOWED, NULL, NULL);
```

Comment:
It would be good to add a info message in the disallowed response
```suggestion
    SendResponse(false, mobile_apis::Result::DISALLOWED, "Service disallowed by policies", NULL);
```

Reply:
Could probably be more specific about what whether its the service name, type or handled rpcs that are disallowed. Not necessary though